### ISSUE TITLE
Memory Leak (Valgrind output)

### ISSUE THREAD

**Issue by jessesimpson36** [htop-dev/htop#908]
Hey everyone,

I noticed that after leaving htop running most of the day, the htop process had consumed over 6GB of memory. I compiled  ran it under valgrind, and found the following:

on commit (442c1596f658daa667e5e25d9bc44beb4072bb64):

```
./configure > configure.txt 2>&1
```
[configure.txt](https://github.com/htop-dev/htop/files/7799444/configure.txt)

```
make debug > make-debug.txt 2>&1 
```
[make-debug.txt](https://github.com/htop-dev/htop/files/7799446/make-debug.txt)

```
valgrind --leak-check=full --show-leak-kinds=all ./htop 2> output.txt
```
[output.txt](https://github.com/htop-dev/htop/files/7799443/output.txt)

I suspect that this is an issue with libncursesw  rather than htop itself.  Can somebody confirm?  And are there any versions of libncursesw without this particular leak?  I can't seem to find that library's git repo.

**Comment by cgzones** [comment]
```
==34511== 595,840 bytes in 76 blocks are still reachable in loss record 78 of 78
==34511==    at 0x483DD99: calloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==34511==    by 0x487F520: newwin_sp (in /usr/lib/x86_64-linux-gnu/libncursesw.so.6.2)
==34511==    by 0x488376B: ??? (in /usr/lib/x86_64-linux-gnu/libncursesw.so.6.2)
==34511==    by 0x487EC0F: newterm_sp (in /usr/lib/x86_64-linux-gnu/libncursesw.so.6.2)
==34511==    by 0x487F08C: newterm (in /usr/lib/x86_64-linux-gnu/libncursesw.so.6.2)
==34511==    by 0x487ADBF: initscr (in /usr/lib/x86_64-linux-gnu/libncursesw.so.6.2)
==34511==    by 0x11F6A9: CRT_init (CRT.c:905)
==34511==    by 0x11DAA0: CommandLine_run (CommandLine.c:345)
==34511==    by 0x119601: main (htop.c:15)
```
These seem to be indeed leaks caused by libncurses, which I have not yet seen. Since running htop 24/7 myself never showed such issue, I presume your local libncursesw version is the culprit.
A snapshot mirror can be found at https://github.com/ThomasDickey/ncurses-snapshots.

**Comment by BenBE** [comment]
Another mirror seems to be: https://github.com/mirror/ncurses

**Comment by jessesimpson36** [comment]
I found a memory leak in my patch/fork, and I must have made an error when I was testing versions of htop that did not have my patch.

I tried running htop for another few hours, this time without my patch, and I found that the memory usage stayed around the same.  If there is an issue with ncurses having a memory leak, it's not so bad that it causes an issue running for a full day.

This issue may be closed.
